src/global/url.ml

```diff
@@ -56,17 +56,29 @@ let minor v =
   | _ -> invalid_arg (v ^ ": invalid  version")
 
 let patch v =
-  match String.split_on_char '.' v with
-  | x :: y :: z :: _ -> x ^ "." ^ y ^ "." ^ z
-  | x :: y :: [] -> x ^ "." ^ y
-  | _ -> invalid_arg (v ^ ": invalid  version")
+  let pre, version = 
+    match String.split_on_char '.' v with
+    | x :: y :: z :: _ -> false, x ^ "." ^ y ^ "." ^ z
+    | x :: y :: [] -> 
+      (match String.split_on_char '+' y with
+      | [] -> assert false
+      | [y] -> false, x ^ "." ^ y
+      | y :: _ -> true, x ^ "." ^ y)
+    | _ -> invalid_arg (v ^ ": invalid  version")
+  in
+  if pre then "v" ^ version
+  else "V" ^ version
```

Comment:
Note that in theory, the tag version (starting with `V`) would work fine for the rc as well: https://rocq-prover.org/doc/V9.0+rc1/refman/index.html exists for instance.

The logic of calling the `patch` function could have been replaced by no call at all.

That being said, I am fine with considering that we link to a branch while in pre-release and to a tagged version afterward.

Note though that so far, we only updated links to the reference manual when releasing the final .0 version. The question then is whether we consider that this is an exceptional case (since the rocq-prover.org website is still in preview) or whether it will keep beeing that way (switching at the time of the rc for 9.1 as well).

Reply:
Yes, there was an issue with + in urls which I was avoiding this way. Superseded by #83 where we point exactly where we want.